From: andrea@suse.de
Subject: low-latency stuff


My point is that preempt and no-preempt should do the same thing there,
otherwise when you benchmark -preempt, you'll get better latency,
but not because of the preempt feature, but just because of unrelated
latency improvements that have nothing to do with preempt.


diff -purN ../linux-2.6.11-rc5.orig/mm/memory.c ./mm/memory.c
--- ../linux-2.6.11-rc5.orig/mm/memory.c	2005-02-24 17:40:15.000000000 +0100
+++ ./mm/memory.c	2005-02-26 16:29:51.468704024 +0100
@@ -602,12 +602,12 @@ static void unmap_page_range(struct mmu_
 	tlb_end_vma(tlb, vma);
 }
 
-#ifdef CONFIG_PREEMPT
-# define ZAP_BLOCK_SIZE	(8 * PAGE_SIZE)
-#else
-/* No preempt: go for improved straight-line efficiency */
-# define ZAP_BLOCK_SIZE	(1024 * PAGE_SIZE)
-#endif
+#ifdef CONFIG_SMP
+/* zap one pte page at a time */
+#define ZAP_BLOCK_SIZE		(FREE_PTE_NR * PAGE_SIZE)
+#else
+#define ZAP_BLOCK_SIZE		(253 * PAGE_SIZE)
+#endif
 
 /**
  * unmap_vmas - unmap a range of memory covered by a list of vma's
